src/frontend/src/lib/services/reward-code.services.ts

```diff
@@ -0,0 +1,63 @@
+import type { VipReward } from '$declarations/rewards/rewards.did';
+import {
+	claimVipReward as claimVipRewardApi,
+	getNewVipReward as getNewVipRewardApi,
+	getUserInfo as getUserInfoApi
+} from '$lib/api/reward.api';
+import { i18n } from '$lib/stores/i18n.store';
+import type { Identity } from '@dfinity/agent';
+import { fromNullable } from '@dfinity/utils';
+import { get } from 'svelte/store';
+
+export const isVipUser = async ({
+	identity,
+	certified
+}: {
+	identity: Identity;
+	certified: boolean;
```

Comment:
Thinking at it now, what's the idea here, it's the UI that decides if it's a certified call or not? You are not planing to use `queryAndUpdate` - which in that case would probably better implemented within the service - or the security does not matter here and were are find with a query?